pkg/daemon/daemon.go

```diff
@@ -492,6 +504,27 @@ func (d *UnikDaemon) initialize() {
 				mountPoints = strings.Split(mntStr, ",")
 			}
 
+			sizeStr := req.FormValue("size")
+			size, err := unikos.ParseSize(sizeStr)
+			if err != nil {
+				return nil, http.StatusBadRequest, errors.New("failed to parse --size argument", err)
+			}
+
+			// FIT SIZE TO FLAVOR
+			// If we are building qemu image for openstack, then we increase value of --size to match the
+			// closest flavor. Why? Because user expects image to be able to consume all the HDD space
+			// that flavor offers (when she runs it on OpenStack).
+			if providerName == openstack_provider && compilerName == compilers.OSV_DYNAMIC_OPENSTACK {
```

Comment:
I agree, this is ugly. But I couldn't find another way to do it since the final size needs to be known prior compiler (Docker container) does it work.